gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/replication/ConfigBasedDataset.java

```diff
@@ -177,12 +231,30 @@ public String datasetURN() {
         watermarkMetadataCopied = true;
       }
 
-      // skip copy same file
-      if (copyToFileMap.containsKey(newPath) && copyToFileMap.get(newPath).getLen() == originFileStatus.getLen()
-          && copyToFileMap.get(newPath).getModificationTime() > originFileStatus.getModificationTime()) {
-        log.debug("Copy from timestamp older than copy to timestamp, skipped copy {} for dataset with metadata {}",
-            originFileStatus.getPath(), this.rc.getMetaData());
-      } else {
+
+      boolean shouldCopy = true;
+      if (copyToFileMap.containsKey(newPath)) {
+        if (this.srcDataFileVersionStrategy.get() instanceof ModTimeDataFileVersionStrategy) {
+          // default version strategy should also compare data file length
+          if (copyToFileMap.get(newPath).getLen() == originFileStatus.getLen()
```

Comment:
I think the file length should be a different comparison altogether, which can be enabled or disabled independently.

Reply:
@ibuenros Let's assume you have the same file copied from source to destination. Now what if you apply different purger algorithm on both src and dst file? I think then the file size would be different, but you don't want to copy again I guess? So should our Purger version strategy exclude the length check?

Reply:
Change a little bit.